src/main/java/org/broadinstitute/hellbender/utils/bwa/BwaMemAlignmentUtils.java

```diff
@@ -15,34 +13,38 @@
 public class BwaMemAlignmentUtils {
     /**
      * Transforms an unaligned GATKRead into an aligned one, as specified by the supplied BwaMemAlignment.
+     * qualsArg can be null.
+     * readGroup can be null.
      *
      * Also, BWA does this odd thing, supplying AS and XS tags for unaligned reads.
      * To produce SAM records that are byte-identical to the ones created by the bwa mem command line,
      * call this method with alwaysGenerateASandXS=true.
      */
-    public static GATKRead applyAlignment(final GATKRead unalignedRead, final BwaMemAlignment alignment,
-                                            final List<String> refNames, final SAMFileHeader header,
-                                            final boolean softclipAlts, final boolean alwaysGenerateASandXS ) {
+    public static SAMRecord applyAlignment( final String readName, final byte[] basesArg, final byte[] qualsArg,
```

Comment:
How come this now returns a SAMRecord instead of a GATKRead? The comment "Transforms an unaligned GATKRead into an aligned one, as specified by the supplied BwaMemAlignment" implies to me that it would return another GATKRead. I'd at least update the comment and maybe add an explanation.